libraries/chain/controller.cpp

```diff
@@ -928,6 +928,39 @@ struct controller_impl {
              // Read-write tasks are not being executed.
    };
 
+   /**
+    *  Plugins / observers listening to signals emitted might trigger
+    *  errors and throw exceptions. Unless those exceptions are caught it could impact consensus and/or
+    *  cause a node to fork.
+    *
+    *  If it is ever desirable to let a signal handler bubble an exception out of this method
+    *  a full audit of its uses needs to be undertaken.
+    *
```

Comment:
Add comments why some exceptions are re-thrown and some are silenced?

Reply:
Updated comment.